src/app/tests/integration/chip_im_responder.cpp

```diff
@@ -104,17 +123,8 @@ void DispatchSingleClusterCommand(const ConcreteCommandPath & aRequestCommandPat
     {
         printf("responder constructing command data in command");
 
-        chip::TLV::TLVWriter * writer;
-
-        const CommandHandler::InvokeResponseParameters prepareParams(aRequestCommandPath);
-        ReturnOnFailure(apCommandObj->PrepareInvokeResponseCommand(path, prepareParams));
-
-        writer = apCommandObj->GetCommandDataIBTLVWriter();
-        ReturnOnFailure(writer->Put(chip::TLV::ContextTag(kTestFieldId1), kTestFieldValue1));
-
-        ReturnOnFailure(writer->Put(chip::TLV::ContextTag(kTestFieldId2), kTestFieldValue2));
-
-        ReturnOnFailure(apCommandObj->FinishCommand());
+        TestTLVDataEncoder testData;
+        apCommandObj->AddResponse(path, kTestCommandId, testData);
```

Comment:
So here's a question....  AddResponse will always need to take a thing that encodes a struct start/end, no?  Seems like we should consider a `StructEncodableToTLV` thing which has a single impl of those and then calls some other virtual method it calls that can be overridden to encode the struct fields...  Not part of this PR, but both this and the network commissioning bits could use it.